zmq/backend/cffi/socket.py

```diff
@@ -299,6 +299,60 @@ def send(self, data, flags=0, copy=False, track=False):
                 frame.close()
             return tracker
 
+    def send_multipart(self, msg_parts, flags=0, copy=True, track=False):
+        """Send a sequence of buffers as a multipart message.
+
+        The zmq.SNDMORE flag is added to all msg parts before the last.
+
+        Parameters
+        ----------
+        msg_parts : iterable
+            A sequence of objects to send as a multipart message. Each element
+            can be any sendable object (Frame, bytes, buffer-providers)
+        flags : int, optional
+            Any valid flags for :func:`Socket.send`.
+            SNDMORE is added automatically for frames before the last.
+        copy : bool, optional
+            Should the frame(s) be sent in a copying or non-copying manner.
+            If copy=False, frames smaller than self.copy_threshold bytes
+            will be copied anyway.
+        track : bool, optional
+            Should the frame(s) be tracked for notification that ZMQ has
+            finished with it (ignored if copy=True).
+
+        Returns
+        -------
+        None : if copy or not track
+        MessageTracker : if track and not copy
+            a MessageTracker object, whose `pending` property will
+            be True until the last send is completed.
+        """
+        # typecheck parts before sending:
+        MsgPartTypes = (Frame, bytes, memoryview)
```

Comment:
shouldnt this be only Frame a specified by the ZMQ specs?

Reply:
Which specs?

C++ send accepts only frames, but the python implementation accepts any bufferable. These 3 are built-in bufferables that don't need to go through a memoryview so they have a fast-path.